cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -98,6 +98,59 @@ def counts_to_results(
     return result
 
 
+def _validate_cirq_circuits(circuits: object) -> None:
+    """Validates that the input is either a single `cirq.Circuit` or a list of `cirq.Circuit`
+    instances.
+
+    Args:
+        circuits: The circuit to run.
+
+    Raises:
+        ValueError: If the input is not a `cirq.Circuit` or a list of
+        `cirq.Circuit` instances.
+    """
+
+    if not (
+        isinstance(circuits, cirq.Circuit)
+        or (
+            isinstance(circuits, Sequence)
+            and all(isinstance(circuit, cirq.Circuit) for circuit in circuits)
+        )
+    ):
+        raise ValueError(
+            "Invalid 'circuits' input. Must be a `cirq.Circuit` or a "
+            "sequence of `cirq.Circuit` instances."
+        )
+
+
+def _validate_integer_param(integer_param: object) -> None:
+    """Validates that an input parameter is positive
+    and an integer.
+
+    Args:
+        integer_param: An input parameter.
+
+    Raises:
+        TypeError: If input is not an integer.
+        ValueError: If input is negative.
+    """
+
+    if not (
+        hasattr(integer_param, "__int__")
+        and int(integer_param) == integer_param
+        or (
+            isinstance(integer_param, (bytes, str))
+            and (
+                str(integer_param, "utf-8") if isinstance(integer_param, bytes) else integer_param
+            ).isdecimal()
+        )
```

Comment:
ah sorry didn't realize bytes.isdecimal doesn't work in python 3.7. let's just go back to
```suggestion
        or (isinstance(integer_param, str) and integer_param.isdecimal())
```